### PR TITLE
TYP: follow-ups to recent PRs

### PR DESCRIPTION
**Pull request by jbrockmendel**

### CODE REVIEW

File: pandas/tests/plotting/frame/test_frame.py [hunk 1]
```diff
@@ -156,8 +156,8 @@ def test_nullable_int_plot(self):
                 "A": [1, 2, 3, 4, 5],
                 "B": [1.0, 2.0, 3.0, 4.0, 5.0],
                 "C": [7, 5, np.nan, 3, 2],
-                "D": pd.to_datetime(dates, format="%Y"),
-                "E": pd.to_datetime(dates, format="%Y", utc=True),
+                "D": pd.to_datetime(dates, format="%Y").view("i8"),
```
Comment (jreback) [inline pandas/tests/plotting/frame/test_frame.py:159]: why this change?

File: pandas/core/internals/blocks.py [hunk 2]
```diff
@@ -1274,22 +1274,6 @@ def shift(self, periods: int, axis: int = 0, fill_value=None):
 
         return [self.make_block(new_values)]
 
-    def _maybe_reshape_where_args(self, values, other, cond, axis):
```
Comment (jreback) [inline pandas/core/internals/blocks.py:1277]: wow!

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/plotting/frame/test_frame.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/tests/plotting/frame/test_frame.py:159]: bc we pass `dtype=np.int64` to the DataFrame constructor (seeing this motivated #38842) which casts these two columns, which is deprecated

File: pandas/core/internals/blocks.py [hunk 3]
```diff
@@ -2583,7 +2567,7 @@ def _block_shape(values: ArrayLike, ndim: int = 1) -> ArrayLike:
     return values
 
 
-def safe_reshape(arr, new_shape: Shape):
+def safe_reshape(arr: ArrayLike, new_shape: Shape) -> ArrayLike:
```
Comment (simonjayhawkins) [inline pandas/core/internals/blocks.py:2570]: The docstring implies that Series and Index are allowed

Comment (jbrockmendel) [reply pandas/core/internals/blocks.py:2559]: updated the docstring

**Review by jreback — APPROVED** [review]
